WooCommerce/Classes/POS/Presentation/PointOfSaleEntryPointView.swift

```diff
@@ -15,26 +19,36 @@ struct PointOfSaleEntryPointView: View {
          collectOrderPaymentAnalyticsTracker: POSCollectOrderPaymentAnalyticsTracking) {
         self.onPointOfSaleModeActiveStateChange = onPointOfSaleModeActiveStateChange
 
-        let posModel = PointOfSaleAggregateModel(
-            itemsController: itemsController,
-            cardPresentPaymentService: cardPresentPaymentService,
-            orderController: orderController,
-            collectOrderPaymentAnalyticsTracker: collectOrderPaymentAnalyticsTracker)
-
-        self._posModel = State(wrappedValue: posModel)
+        self.itemsController = itemsController
+        self.cardPresentPaymentService = cardPresentPaymentService
+        self.orderController = orderController
+        self.collectOrderPaymentAnalyticsTracker = collectOrderPaymentAnalyticsTracker
     }
 
     var body: some View {
-        PointOfSaleDashboardView()
+        Group {
+            if let posModel = posModel {
+                PointOfSaleDashboardView()
+                    .environment(posModel)
+            } else {
+                PointOfSaleLoadingView()
+            }
+        }
+        .task {
+            posModel = PointOfSaleAggregateModel(
```

Comment:
This pattern is so unexpected (at least for me), that I would even paste a link to Apple docs here for future reference / add an explanation

Reply:
Good point, added!